### PR TITLE
Feature/advanced graphs

### PR DESCRIPTION
**Pull request by netashefer**

### CODE REVIEW

File: src/components/Common/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
 type DropdownProps = {
-	value: FieldConfig;
+	value: { label: string, value: string; };
 	onChange: Dispatch<SetStateAction<any>>;
 	label: string;
-	items: FieldConfig[];
+	items: Item[];
+};
+export type Item<T = string> = {
```
Comment (YahelR) [inline src/components/Common/Dropdown/Dropdown.tsx]: this type can't be used for the value?

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 2]
```diff
@@ -7,29 +7,33 @@ import { useAddWidget } from "../../../recoil/customHooks/useWidgetHandler";
...
+import { chartMapping, graphTypes } from "./chartParametersMapping";
 import './ParametersPanel.scss';
 
 const ParametersPanel = () => {
 	const dataSources = useRecoilValue(dataSourcesAtom);
-	const [dataSource, setDataSource] = useState<FieldConfig>(null);
-	const [graphType, setGraphType] = useState<FieldConfig>({ value: 'pie', displayName: 'Pie' }); //for the time being all of these are useState
-	const [xAxis, setXAxis] = useState<FieldConfig>(null);
-	const [yAxis, setYAxis] = useState<FieldConfig>(null);
+	const [dataSource, setDataSource] = useState<Item>(null);
```
Comment (YahelR) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: item as a name feels a bit too generic

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 3]
```diff
@@ -46,37 +50,83 @@ const ParametersPanel = () => {
...
+				x_field: xAxis?.value || null,
+				y_field: { aggragation: yAxis?.value, field: yAxisField?.value || null },
+				dataFields: dataFields?.map(f => ({ field: f, label: f }))
 			},
 			template: { type: graphType.value as GraphType },
 			title: graphName,
 		};
 
-		try{
+		console.log(graphToSave);
```
Comment (YahelR) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: no no + whitelines

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 4]
```diff
@@ -46,37 +50,83 @@ const ParametersPanel = () => {
...
+				label="Data Source"
+				items={dataSources?.map(ds => ({ label: ds.displayName, value: ds.dataSourceId }))}
+			/>
+			<Dropdown
+				value={graphType}
+				onChange={setGraphType}
+				label="Graph Type"
+				items={graphTypes}
+			/>
+			<RenderIf condition={!!(graphType && dataSource && chartConfig)}>
```
Comment (YahelR) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx:88]: I like this render if, looks clean!

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 5]
```diff
@@ -46,37 +50,83 @@ const ParametersPanel = () => {
...
+				<>
+					<RenderIf condition={!!chartConfig?.yFieldOptions?.length}>
+						<Dropdown
+							value={xAxis}
+							onChange={setXAxis}
+							label={chartConfig?.xFieldLabel || "X Axis"}
+							items={schemaFields}
+						/>
+					</RenderIf>
+					<RenderIf condition={!!chartConfig?.yFieldOptions?.length}>
```
Comment (YahelR) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: isnt it the same condition as the one before it? why not combine them to be both under it

File: src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx [hunk 6]
```diff
@@ -46,37 +50,83 @@ const ParametersPanel = () => {
...
+						<Dropdown
+							value={yAxis}
+							onChange={setYAxis}
+							label="Y Axis"
+							items={chartConfig?.yFieldOptions?.map(v => ({ label: v.funcDisplayName, value: v.func }))}
+						/>
+					</RenderIf>
+					<RenderIf condition={yAxis?.value === 'uniqueValues'}>
```
Comment (YahelR) [inline src/components/GraphWorkshop/ParametersPanel/ParametersPanel.tsx]: maybe we should consider an enum for the yaxis values

File: src/components/MultiSelect/MultiSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    const handleChange = (event: SelectChangeEvent<typeof selectedOptions>) => {
+        const { target: { value } } = event;
+        setSelectedOptions(
+            // On autofill we get a stringified value.
+            typeof value === 'string' ? value.split(',') : value,
+        );
+    };
+
+    return (
+        <FormControl sx={{ m: 1, width: 300 }}>
```
Comment (YahelR) [inline src/components/MultiSelect/MultiSelect.tsx:48]: cool, but this is kind of convention breaking

but - I don't really care and I honestly think they won't either, so leave it be

Comment (netashefer) [reply src/components/MultiSelect/MultiSelect.tsx:48]: copied from google :)

File: src/components/Common/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (netashefer) [reply src/components/Common/Dropdown/Dropdown.tsx]: what do you mean?

**Pull request merged**